source/solvers/heat_transfer.cc

```diff
@@ -969,7 +1019,780 @@ HeatTransfer<dim>::solve_linear_system(const bool initial_step,
+  this->statistics_table.add_value("max", maximum_temperature);
+  this->statistics_table.add_value("average", temperature_average);
+  this->statistics_table.add_value("std-dev", temperature_std_deviation);
+}
+
+template <int dim>
+void
+HeatTransfer<dim>::write_temperature_statistics(const std::string domain_name)
+{
+  auto mpi_communicator = triangulation->get_communicator();
+
+  if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+    {
+      std::string filename =
+        simulation_parameters.simulation_control.output_folder +
+        simulation_parameters.post_processing.temperature_output_name + "_" +
+        domain_name + ".dat";
+      std::ofstream output(filename.c_str());
+
+      this->statistics_table.write_text(output);
+    }
+}
+
+template <int dim>
+template <typename VectorType>
+void
+HeatTransfer<dim>::postprocess_heat_flux_on_bc(
+  const bool        gather_vof,
+  const VectorType &current_solution_fd)
+{
+  const unsigned int n_q_points_face  = this->face_quadrature->size();
+  const MPI_Comm     mpi_communicator = this->dof_handler.get_communicator();
+
+  // Initialize heat transfer information
+  std::vector<double>         local_temperature_values(n_q_points_face);
+  std::vector<Tensor<1, dim>> temperature_gradient(n_q_points_face);
+  FEFaceValues<dim>           fe_face_values_ht(*this->temperature_mapping,
+                                      this->dof_handler.get_fe(),
+                                      *this->face_quadrature,
+                                      update_values | update_gradients |
+                                        update_JxW_values |
+                                        update_normal_vectors);
+
+  // Initialize fluid dynamics information
+  const DoFHandler<dim> *dof_handler_fd =
+    multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+  const FEValuesExtractors::Vector velocities(0);
+  std::vector<Tensor<1, dim>>      local_velocity_values(n_q_points_face);
+  FEFaceValues<dim>                fe_face_values_fd(*this->temperature_mapping,
+                                      dof_handler_fd->get_fe(),
+                                      *this->face_quadrature,
+                                      update_values);
+
+  // Initialize VOF information
+  const DoFHandler<dim> *            dof_handler_vof;
+  std::shared_ptr<FEFaceValues<dim>> fe_face_values_vof;
+  std::vector<double>                phase_values(n_q_points_face);
+
+  if (gather_vof)
+    {
+      dof_handler_vof = this->multiphysics->get_dof_handler(PhysicsID::VOF);
+      fe_face_values_vof =
+        std::make_shared<FEFaceValues<dim>>(*this->temperature_mapping,
+                                            dof_handler_vof->get_fe(),
+                                            *this->face_quadrature,
+                                            update_values);
+    }
+
+  // Initialize fluid properties
+  auto properties_manager =
```

Comment:
I would make it a reference though
```suggestion
  auto & properties_manager =
```